docker/letsencrypt/docker-entrypoint.sh

```diff
@@ -18,9 +18,11 @@ set +e
 if [ "$LETSENCRYPT_MODE" == "staging" ]; then
     printf "\nTrying to get STAGING certificate\n"
     certbot --config-dir "/geonode-certificates/$LETSENCRYPT_MODE" certonly --webroot -w "/geonode-certificates" -d "$HTTPS_HOST" -m "$ADMIN_EMAIL" --agree-tos --non-interactive --test-cert
+    docker ps --filter name=nginx4${COMPOSE_PROJECT_NAME} --filter status=running -aq | xargs docker restart
```

Comment:
That would break Kubernetes deployments, and exposing the Docker socket to a service can be a security issue (and sometimes it is not possible).

Reply:
@frafra I do not think that letsencrypt container or this nginx container would work at all with kubernetes, and I imagine I would run geonode + celery + rabbit + postgres served by an ingress so the nginx container would not be used. Concerning letsencrypt on k8s there is plenty of other ways to be pursued.

Reply:
@lpasquali you are right that there are better ways to handle letsencrypt on Kubernetes.
I still think that this solution can have issues:
1. A container should not have the ability to restart the others usually
2. That ties GeoNode to Docker (no Podman, no Kubernetes) and requires access to its socket (which is not permitted on many production system, as that can be a security issue)

Wouldn't it be easier to check the letsencrypt flag in nginx entrypoint to set the path for the certificate?

Reply:
Hello @frafra I do not think that letsencrypt container or this nginx container would work at all with kubernetes, and I imagine I would run geonode + celery + rabbit + postgres served by an ingress so the nginx container would not be used. Concerning letsencrypt on k8s there is plenty of other ways to be pursued.
Regarding security, I see your point but I would also remind that there is no services exposed by the letsencrypt container itself

Reply:
Thank you for your reply. I think that the same comment has been posted twice.
We can argue about security as developers, but sysadmins are not, so some systems are just denying the usage of the docker socket (or they do not have that entirely by default, like podman).
I do not want to waste your time, so mine are just suggestions, but I do not see why such an antipattern should be used instead of having nginx starting with the right certificates in the first place. If there is a time issue in writing such PR, I can help with that.

Reply:
@frafra I removed usage of docker in docker, check my last commit so you can find how to contribute next time with useful code: I took the only relevant/practical thing of your statements (the security issue) and implemented solution within the existing scripts.